experiments/storm/src/main/scala/io/gearpump/experiments/storm/topology/GearpumpStormComponent.scala

```diff
@@ -113,14 +121,20 @@ object GearpumpStormComponent {
       val dag = getDAG(appMaster)
       val topologyContext = getTopologyContext(dag, taskId)
       collector = getOutputCollector(taskContext, topologyContext)
-      val delegate = new StormSpoutOutputCollector(collector)
-      spout.open(config, topologyContext, new SpoutOutputCollector(delegate))
+      spout.open(config, topologyContext, new SpoutOutputCollector(collector))
     }
 
     override def next(message: Message): Unit = {
-      collector.setTimestamp(System.currentTimeMillis())
       spout.nextTuple()
     }
+
+    def onCheckpointClock(clock: TimeStamp): Unit = {
+      collector.ack(clock)
+    }
+
+    private def isSequentiallyReplayableSpout(className: String): Boolean = {
```

Comment:
kafka is the only external source module in storm-0.9.x version. There could be more when we upgrade to storm-0.10.